example/stateful_provider.dart

```diff
@@ -23,8 +23,8 @@ class Bloc {
 class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return StatefulProvider<Bloc>(
-      valueBuilder: (_) => Bloc(),
+    return Provider<Bloc>(
```

Comment:
[nit] Can we change the names of these files? Since the "default" is now `Provider()` (and not `Provider.value()`) I'd be for either merging the two files or doing something like `provider.dart` (for stateful)` and `provider_value.dart` (for `Provider.value()`).